File: autox/src/autox/actions/Enter.java
package autox.actions;

import org.openqa.selenium.WebElement;

/**
 * Created with AutoX project.
 * User: jien.huang
 * Date: 12/15/12
 */
public class Enter extends Action {
    @Override
    public Object findTestObject() {
        return findUIObject();
    }

    @Override
    protected void handle(Object testObject) {
        if(testObject==null){
            getResult().Error("The target want to set text to not found!");
        }
        String data = getAttributeData();
        //TODO need to handle file upload field, it cannot be cleared.
        ((WebElement)testObject).clear();
        ((WebElement)testObject).sendKeys(data);
    }

}

File: dotnet/AutoX.Client/Enter.cs
// Hapa Project, CC
// Created @2012 08 24 09:25
// Last Updated  by Huang, Jien @2012 08 24 09:25

#region

using System.Xml.Linq;
using OpenQA.Selenium;

#endregion

namespace AutoX.Client
{
    internal class Enter : AbstractAction
    {
        public override XElement Act()
        {
            var sr = new StepResult(this);
            if (UIObject.Count == 0)
            {
                sr.Error("Expected UI Object is not found!");
            }
            else
            {
                IWebElement text = UIObject[0];
                text.Clear();
                if (!string.IsNullOrEmpty(Data))
                {
                    text.SendKeys(Data);
                }
            }
            return sr.GetResult();
        }
    }
}